airbyte-integrations/connectors/source-hubspot/source_hubspot/helpers.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import abc
+from typing import MutableMapping
+
+
+class IRecordPostProcessor(abc.ABC):
+    @abc.abstractmethod
+    def add_record(self, record: MutableMapping, primary_key: str = None):
+        """"""
+
+    @property
+    @abc.abstractmethod
+    def flat(self):
+        """"""
+
+
+class GroupByKey(IRecordPostProcessor):
+    def __init__(self):
+        self._storage = {}
+
+    def add_record(self, record: MutableMapping, primary_key: str = None):
```

Comment:
nit: I think we can set the primary key in the constructor. that'll clean up the interface since StoreAsIs doesn't need to know the primary key

Reply:
:+1: done